Nonprofit-Exchange-Hub/web-app
# 152 asset dtos should not include user

## Dev Summary

Removing the `user: User` props from the assets dto and implementing authed routes to
- protect endpoint
- extract user from session

Only doing assets for now. I added another ticket for messages

Includes changes to tests and seeding

### changes not related to ticket, but to tooling
- added a jest config file to separate the unit test configs in `src` separate from the e2e/integration test configs in `test`. This change was needed as the `npm run test` was also picking up the tests in the `test` project




## Test Plan

See integration test for test cases and further steps
repro steps:
1. log in with insomnia or postman
2. save your cookie in your http client
3. make post request to `assets` with `json` body. If your http client is not sending the cookie from your log in request, grab it from the login response and add an http header for `Cookie` and for the value add the cookie the server returned on the log in request.

Expect to see an asset created without passing a user in the body of the request
Try it again without logging in or passing a cookie in the request and your should get a forbidden response



## Resources
